lua/starfall/libs_cl/vgui.lua

```diff
@@ -0,0 +1,2782 @@
+	local uwp = unwrap(self)
+	
+	uwp:SetPos(x, y)
+end
+
+--- Returns the position of the panel relative to its Panel:getParent.
+--@return number X coordinate, relative to this panels parents top left corner.
+--@return number Y coordinate, relative to this panels parents top left corner.
+function pnl_methods:getPos()
+	local uwp = unwrap(self)
+	
+	return uwp:GetPos()
+end
+
+--- Returns the value the panel holds.
+--- In engine is only implemented for CheckButton, Label and TextEntry as a string.
+--@param any The value the panel holds.
+function pnl_methods:getValue()
+	local uwp = unwrap(self)
+
+	return uwp:GetValue()
+end
+
+--- Sets the size of the panel.
+--@param number x Width of the panel.
+--@param number y Height of the panel.
+function pnl_methods:setSize(x, y)
+	checkluatype(x, TYPE_NUMBER)
+	checkluatype(y, TYPE_NUMBER)
+	local uwp = unwrap(self)
+	
+	uwp:SetSize(x, y)
+end
+
+--- Returns the size of the panel.
+--@return number width
+--@return number height
+function pnl_methods:getSize()
+	local uwp = unwrap(self)
+	
+	return uwp:GetSize()
+end
+
+--- Sets the height of the panel.
+--@param number newHeight The height to be set.
+function pnl_methods:setHeight(val)
+	checkluatype(val, TYPE_NUMBER)
+	local uwp = unwrap(self)
+
+	uwp:SetHeight(val)
+end
+
+--- Gets the height of the panel.
+--@return number The height of the panel.
+function pnl_methods:getHeight()
+	local uwp = unwrap(self)
+
+	return uwp:GetTall()
+end
+
+--- Sets the width of the panel.
+--@param number newWidth The width to be set.
+function pnl_methods:setWidth(val)
+	checkluatype(val, TYPE_NUMBER)
+	local uwp = unwrap(self)
+
+	uwp:SetWidth(val)
+end
+
+--- Gets the width of the panel.
+--@return number The width of the panel.
+function pnl_methods:getWidth()
+	local uwp = unwrap(self)
+
+	return uwp:GetWide()
+end
+
+--- Sets the text value of a panel object containing text, such as DLabel, DTextEntry or DButton.
+--@param string text The text value to set.
+function pnl_methods:setText(text)
+	checkluatype(text, TYPE_STRING)
+	local uwp = unwrap(self)
+	
+	uwp:SetText(text)
+end
+
+--- Sets the tooltip to be displayed when a player hovers over the panel object with their cursor.
+--@param string text The text to be displayed in the tooltip.
+function pnl_methods:setTooltip(text)
+	checkluatype(text, TYPE_STRING)
+	local uwp = unwrap(self)
+
+	uwp:SetTooltip(text)
+end
+
+--- Removes the tooltip on the panel set with Panel:setTooltip
+function pnl_methods:unsetTooltip()
+	local uwp = unwrap(self)
+
+	uwp:SetTooltip(false)
+end
+
+--- Sets the panel to be displayed as contents of a DTooltip when a player hovers over the panel object with their cursor.
+--@param Panel The panel to use as the tooltip.
+function pnl_methods:setTooltipPanel(setPnl)
+	local uwp = unwrap(self)
+	local uwsp = pnlunwrap(setPnl)
+
+	uwp:SetTooltipPanel(uwsp)
+end
+
+--- Removes the tooltip panel set on this panel with Panel:setTooltipPanel
+function pnl_methods:unsetTooltipPanel()

```

Comment:
I don't think this should exist, just have the panel be optional in `setTooltipPanel`, just like how starfall handles `setParent()` with nil.

Reply:
Done